client/landing/gutenboarding/index.tsx

```diff
@@ -14,23 +18,45 @@ import { Gutenboard } from './gutenboard';
 import { setupWpDataDebug } from './devtools';
 import accessibleFocus from 'lib/accessible-focus';
 import { path, Step } from './path';
+import { USER_STORE } from './stores/user';
 
 /**
  * Style dependencies
  */
 import 'assets/stylesheets/gutenboarding.scss';
 import 'components/environment-badge/style.scss';
 
-window.AppBoot = () => {
+const DEFAULT_LOCALE_SLUG: string = config( 'i18n_default_locale_slug' );
+
+type User = import('@automattic/data-stores').User.CurrentUser;
+
+interface AppWindow extends Window {
+	currentUser?: User;
+	i18nLocaleStrings?: string;
+}
+declare const window: AppWindow;
+
+window.AppBoot = async () => {
 	if ( ! config.isEnabled( 'gutenboarding' ) ) {
 		window.location.href = '/';
-	} else {
-		setupWpDataDebug();
+		return;
+	}
+	setupWpDataDebug();
 
-		// Add accessible-focus listener.
-		accessibleFocus();
+	// Add accessible-focus listener.
+	accessibleFocus();
 
-		ReactDom.render(
+	let locale = DEFAULT_LOCALE_SLUG;
+	try {
+		const [ userLocale, localeData ] = await getLocale();
+		console.log( 'Setting locale data: %o', localeData );
+		setLocaleData( localeData );
```

Comment:
@sirreal I think this is a weird race condition with creating a React context. It looks like `setLocaleData` _is_ working, but not working for the `__()` method called from the context reference. As a test, calling `setLocaleData` from within the I8nProvider results in the page translating as expected and seems to support this theory.

Reply:
I did a bit of debugging:

```diff
diff --git a/packages/react-i18n/src/index.tsx b/packages/react-i18n/src/index.tsx
index c7375de4c2..dc88e7f12c 100644
--- a/packages/react-i18n/src/index.tsx
+++ b/packages/react-i18n/src/index.tsx
@@ -26,7 +26,9 @@ interface Props {
 export const I18nProvider: React.FunctionComponent< Props > = ( { children, locale } ) => {
        const [ contextValue, setContextValue ] = React.useState< I18nProps >( makeI18n( locale ) );
        React.useEffect( () => {
-               setContextValue( makeI18n( locale ) );
+               const intl = makeI18n( locale );
+               setContextValue( intl );
+               console.log( 'useEffect', locale, intl.__( 'Back' ) );
        }, [ locale ] );
        return <I18nContext.Provider value={ contextValue }>{ children }</I18nContext.Provider>;
 };
```

gives one occurrence of

> useEffect de Back

('Back' should be 'Zurück')

:thinking: 

This isn't another one of those 'multiple versions of a `@wordpress/` package installed' cases, is it? Cause that would explain why the singleton pattern would fail here :thinking:

Reply:
My top-level `node_modules/` has `@wordpress/i18n@3.8.0`, my `client/node_modules/` has `@wordpress/i18n@3.9.0` :thinking:

Reply:
> My top-level `node_modules/` has `@wordpress/i18n@3.8.0`, my `client/node_modules/` has `@wordpress/i18n@3.9.0` 🤔

I suspect you're right, this is going to be another duplicated singleton.